src/app/routes/connector-ui/contract-agreement-page/contract-agreement-page/contract-agreement-page.service.ts

```diff
@@ -105,4 +127,11 @@ export class ContractAgreementPageService {
       filter((it) => it.isReady),
     );
   }
+
+  private fetchLimits(): Observable<ConnectorLimits | null> {
+    if (this.activeFeatureSet.hasEnterPriseEditionFields()) {
+      return this.edcApiService.getEnterpriseEditionConnectorLimits();
+    }
+    return of(null);
+  }
 }
```

Comment:
There should be a contract agreement detail dialog. The contract agreement's status + info icon + tooltip should also be shown there (might need property grid field adjustment for the icon + tooltip).